lib/spack/spack/build_systems/lua.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+
+from llnl.util.filesystem import find
+
+from spack.directives import depends_on, extends
+from spack.package import PackageBase, run_before
+from spack.util.executable import Executable
+
+
+class LuaPackage(PackageBase):
+    """Specialized class for lua packages"""
+
+    phases = ['unpack', 'preprocess', 'install']
+    #: This attribute is used in UI queries that need to know the build
+    #: system base class
+    build_system_class = 'LuaPackage'
+
+    list_depth = 1  # LuaRocks requires at least one level of spidering to find versions
+    depends_on('lua-lang')
+    extends('lua', when='^lua')
+    extends('lua-luajit', when='^lua-luajit')
+    depends_on('luajit', when='^lua-luajit')
+    depends_on('luajit', when='^lua-luajit-openresty')
+    depends_on('lua-luajit+lualinks', when='^lua-luajit')
+    extends('lua-luajit-openresty', when='^lua-luajit-openresty')
+    depends_on('lua-luajit-openresty+lualinks', when='^lua-luajit-openresty')
```

Comment:
I think this should work too and is a bit more readable:
```suggestion
    with when('^lua-luajit'):
        extends('lua-luajit')
        depends_on('luajit')
        depends_on('lua-luajit+lualinks')
    
    with when('^lua-luajit-openresty'):
        extends('lua-luajit-openresty')
        depends_on('luajit')
        depends_on('lua-luajit-openresty+lualinks')
```